internal/hardware/validator.go

```diff
@@ -273,8 +273,8 @@ func (v *validator) getOCPRequirementsForVersion(cluster *common.Cluster) (*mode
 	return requirements, nil
 }
 
-func compileDiskReasonTemplate(template string, wildcards ...string) *regexp.Regexp {
-	tmp, err := regexp.Compile(fmt.Sprintf(regexp.QuoteMeta(template), wildcards))
+func compileDiskReasonTemplate(template string, wildcards ...interface{}) *regexp.Regexp {
```

Comment:
Why was this change needed? After all the wildcards will always be strings.

Reply:
The compileDiskReasonTemplate is a string variadic function while Sprintf is an Interface variadic function.
We must pass the exact type to the function.
In addition we must expand this array before passing it to the Sprintf method, else Sprintf method would deal with it as a single argument instead of an array of arguments.
The easy path is just changing the slice type to an interface slice and eliminate the explicit conversion (Creating a new slice and copy all the values)

Reply:
In logical terms the result pattern was "Disk is too small /(disk only has [.* .*], but %s(Missing) are required/)" instead of "Disk is too small /(disk only has .* , but .* are required/)"